fvm/src/machine/default.rs

```diff
@@ -92,6 +93,20 @@ where
             ));
         }
 
+        if !blockstore
+            .has(&EMPTY_ARR_CID)
+            .context("failed to check blockstore for empty array CID")?
+        {
+            let empty_cid = blockstore.put_cbor::<[(); 0]>(&[], Blake2b256)?;
+            if empty_cid != *EMPTY_ARR_CID {
+                return Err(anyhow!(
+                    "empty CID sanity check failed, {} != {}",
+                    empty_cid,
+                    *EMPTY_ARR_CID,
+                ));
+            }
+        }
```

Comment:
nit: I'd just put it without checking. It'll likely be _faster_.

Reply:
It'll be faster because:

1. Hashing is pretty fast.
2. Reading from the blockstore is flow.
3. A couple of extra puts likely won't cause any issues (and, to be honest, this block will likely be written by several actors as well.

Reply:
It'll also be more correct. With the way the code is currently written, the splitstore could come in and delete the block.

Reply:
Ack, thanks!